src/overlays/actors/ovl_Bg_Spot18_Obj/z_bg_spot18_obj.c

```diff
@@ -21,47 +41,259 @@ const ActorInit Bg_Spot18_Obj_InitVars = {
     (ActorFunc)BgSpot18Obj_Update,
     (ActorFunc)BgSpot18Obj_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8910.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8A5C.s")
+u8 D_808B90F0[] = {
+    0x01,
+    0x01,
+    0x01,
+    0x00,
+};
+
+f32 D_808B90F4[] = {
+    0.1f,
+    0.1f,
+};
+
+UNK_TYPE D_808B90FC[] = {
+    0x06002FE4,
+    0x0600261C,
+};
+
+u32 D_808B9104[] = {
+    0,
+    0,
+};
+
+BgSpot18ObjInitFunc D_808B910C[] = {
+    func_808B8A98,
+    func_808B8910,
+    func_808B8A5C,
+    func_808B8B08,
+};
+
+static InitChainEntry sInitChain1[] = {
+    ICHAIN_F32(minVelocityY, 65526, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, 65532, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 1400, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 800, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChain2[] = {
+    ICHAIN_F32(unk_F4, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 700, ICHAIN_STOP),
+};
+
+BgSpot18ObjInitFunc D_808B913C[] = {
+    func_808B8BB4,
+    func_808B8C90,
+};
+
+static Gfx* sDlists[] = {
+    0x06002BC0,
+    0x06002370,
+};
+
+s32 func_808B8910(BgSpot18Obj* this, GlobalContext* globalCtx) {
+    s32 isAdult;
+
+    if (LINK_AGE_IN_YEARS == YEARS_ADULT) {
+        isAdult = 1;
+    } else if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+        isAdult = 0;
+    } else {
+        osSyncPrintf("Error : リンク年齢不詳 (%s %d)(arg_data 0x%04x)\n", "../z_bg_spot18_obj.c", 182,
+                     this->dyna.actor.params);
+        return 0;
+    }
+
+    switch (D_808B90F0[((this->dyna.actor.params & 0xF) * 2) + isAdult]) {
```

Comment:
D_808B90F0 is most likely a 2d array based on this code, and imo `isAdult` isn't quite appropriate since the variable is used as an array index here, and not as a boolean.
```suggestion
    switch (D_808B90F0[this->dyna.actor.params & 0xF][isAdult]) {
```

Reply:
I changed it to a 2d array, but I can't think of another decent name. Maybe linkAge?

Reply:
"age" is fine imo